src/pan-view/pan-view.cc

```diff
@@ -967,28 +961,23 @@ static GList *pan_layout_intersect_l(GList *list, GList *item_list,
 GList *pan_layout_intersect(PanWindow *pw, gint x, gint y, gint width, gint height)
 {
 	GList *list = nullptr;
-	GList *grid;
-	PanGrid *pg = nullptr;
 	const GdkRectangle rect{x, y, width, height};
 
-	grid = pw->list_grid;
-	while (grid && !pg)
-		{
-		pg = static_cast<PanGrid *>(grid->data);
-		grid = grid->next;
+	const auto pan_grid_contains_rect = [](gconstpointer data, gconstpointer user_data) -> gint
+	{
+		const auto *pg = static_cast<const PanGrid *>(data);
+		const auto *rect = static_cast<const GdkRectangle *>(user_data);
 
-		if (x < pg->x || x + width > pg->x + pg->w ||
-		    y < pg->y || y + height > pg->y + pg->h)
-			{
-			pg = nullptr;
-			}
-		}
+		GdkRectangle intersection;
+		return (gdk_rectangle_intersect(&pg->rect, rect, &intersection) && gdk_rectangle_equal(rect, &intersection)) ? 0 : 1;
```

Comment:
I would split this across multiple lines just for legibility.  (Also, just to note this for future style conversations, this is a place where I would want to use a single-line `if (...) return 1;` instead of adding the braces)
```
GdkRectangle intersection;
if (!gdk_rectangle_intersect(...))
    {
    return 1;
    }
return gdk_rectangle_equal(...)
```